src/server/publicApi.js

```diff
@@ -1,125 +1,181 @@
+                    let cell = null;
+                    for (let i = 0; i < diagram.cells.length; i++) {
+                        let c = diagram.cells[i];
+                        if (c.id === componentIdx) {
+                            cell = c;
+                            break;
+                        }
+                    }
+                    if (cell !== null) {
+                        let threats = [];
+                        if (Array.isArray(cell.threats)) {
+                            threats = cell.threats;
+                        }
+                        Object.keys(
+                            res.G.identifiedThreats[diagramIdx][componentIdx]
+                        ).forEach((threatIdx) => {
+                            let t =
+                                res.G.identifiedThreats[diagramIdx][componentIdx][threatIdx];
+                            threats.push({
+                                status: "Open",
+                                severity: t.severity,
+                                id: t.id,
+                                methodology: "STRIDE",
+                                type: getTypeString(t.type),
+                                title: t.title,
+                                description: t.description,
+                                mitigation: t.mitigation,
+                                owner: metadata.players[t.owner].name,
+                                game: gameID,
+                            });
+                        });
+                        cell.threats = threats;
+                    }
+                }
+            );
         });
+
+        const modelTitle = model.summary.title.replace(" ", "-");
+        const timestamp = new Date().toISOString().replace(":", "-");
+        ctx.attachment(`${modelTitle}-${timestamp}.json`);
+        ctx.body = model;
     });
 
-    const modelTitle = model.summary.title.replace(' ', '-');
-    const timestamp = (new Date()).toISOString().replace(":", "-");   
-    ctx.attachment(`${modelTitle}-${timestamp}.json`);
-    ctx.body = model;
+
+    //produce a nice textfile with the threats in
+    router.get("/download/text/:id", async (ctx) => {
+        //get some variables that might be useful
+        const gameName = ElevationOfPrivilege.name;
+        const gameID = ctx.params.id;
+        const res = await gameServer.db.get(`${gameName}:${gameID}`);
+        var threats = [];
+
+        Object.keys(res.G.identifiedThreats).forEach((diagramId) => {
+            Object.keys(res.G.identifiedThreats[diagramId]).forEach((componentId) => {
+                Object.keys(res.G.identifiedThreats[diagramId][componentId]).forEach((threatId) => {
+                    threats.push(res.G.identifiedThreats[diagramId][componentId][threatId]);
+                });
+            });
+        });
+        
+        const timestamp = new Date().toISOString().replace(":", "-");
+        ctx.attachment(`threats-${timestamp}.txt`);
```

Comment:
Could you do it like in line 126 and also add the `modelTitle` to the name?  I might need to download this file for several teams and don't want to get confused...  :-P

Reply:
I also removed "threats" from the title because when I downloaded it using "model via a different channel" it ended up being "threats-Threat-Modelling-... .txt" and it's not really needed. Let me know if you think I should keep it in or put it in some other way

Reply:
Actually, I would keep the "threats"-part. "Threat Modelling" from the model file often would be the name of the team or application considered. In this case the file name would be `threats-someAwesomeProject-... .txt` which I find rather descriptive.

Reply:
I've put it back and made it a .md file